### PR TITLE
Fix deprecated Threads.nthreads()

### PR DESCRIPTION
**Pull request by IanButterworth**
`Threads.nthreads()` is going to be deprecated in https://github.com/JuliaLang/julia/pull/45447

This needs to be fixed before https://github.com/JuliaLang/julia/pull/45447 merges because Base tests run with `--depwarn=error` https://github.com/JuliaLang/julia/blob/df846436bb91b291be988f5e18e256d514b97347/test/testenv.jl#L19

### CODE REVIEW

**Review by DilumAluthge — DISMISSED** [review]

**Comment by IanButterworth** [comment]
I think this needs signoff from @JeffBezanson / @vtjnash that `Threads.maxthreadid()` is the name that's going to be used, given https://github.com/JuliaLang/julia/pull/45447 can't be merged first

**Review by vtjnash — CHANGES_REQUESTED** [review]
FWIW, this would be very wrong, since there is no replacement for `nthreads` now

**Comment by Wimmerer** [comment]
That will require a pretty significant rewrite of the CHOLMOD wrapper. We'll have to carry workspaces and settings around with each Factor object now I believe.

**Comment by DilumAluthge** [comment]
Could you use `Base.task_local_storage`?

**Comment by IanButterworth** [comment]
I'll leave the necessary changes up to @Wimmerer

**Pull request closed without merging**
